src/frontend/screens/Quests/components/QuestDetailsViewPlay/index.tsx

```diff
@@ -36,11 +37,38 @@ export function QuestDetailsViewPlayWrapper({
 
   const navigateToGame = useMutation({
     mutationFn: async (appName: string) => {
-      await window.api.addHyperplayGame(appName)
-      const gameInfo = await getGameInfo(appName, 'hyperplay')
-      navigate(`/gamepage/hyperplay/${appName}`, {
-        state: { gameInfo, fromDM: false }
-      })
+      const { appName: epicAppName, epicListingUrl } = await fetchEpicListing(
+        appName
+      )
+      const runner: Runner = epicListingUrl ? 'legendary' : 'hyperplay'
+
+      // check for gameinfo to see if it is on the library
+      return getGameInfo(epicAppName || appName, runner)
```

Comment:
can we set these in the same if statement like
```
let runner = 'hyperplay'
let name = appName
if (epicListingUrl && epicAppName){
  runner = 'legendary
  name = epicAppName
}
```

otherwise if `fetchEpicListing` returns empty string for appName but a correct epic listing url, it would try to navigate to the 'hyperplay' runner with the epic app name

Reply:
Good call

Reply:
Ok, after changing it stoppeed working and now I remember why I did this way. It is because for the last case (navigate to epic store) we need it to be runner legendary BUT has an epic listing as well.
but I can adjust this logic.